lib/logstash/outputs/elasticsearch/http_client/manticore_adapter.rb

```diff
@@ -84,19 +84,19 @@ def perform_request(url, method, path, params={}, body=nil)
     end
 
     def format_url(url, path=nil)
-
-      request_uri = url.uri
+      request_uri = url.clone
 
       if path
         # Combine the paths using the minimal # of /s
         # First, we make sure the path is relative so URI.join does
         # the right thing
         relative_path = path && path.start_with?("/") ? path[1..-1] : path
-        # because URI.join obliterates the query parameter, we need to save it
-        # and restore it after this operation
-        query = request_uri.query
-        request_uri = URI.join(request_uri, relative_path)
-        request_uri.query = query if query
+
+        if !request_uri.path
+          request_uri.path = path
+        else
+          request_uri.path = request_uri.path + "/" + relative_path
```

Comment:
this is on the critical path for requests, so I suggest using string interpolation to create less strings